File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        
        groups = defaultdict(list)
        
        for i in range(len(strs)):
            sorted_str = ''.join(sorted(strs[i]))
            groups[sorted_str].append(strs[i])
            
        print(groups)
            
        return [groups[key] for key in groups.keys()]
        

File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    const groups = {};
    
    strs.forEach(str => {
        const counts = Array(26).fill(0);
        
        for (const char of str) {
            counts[char.charCodeAt(0) - 97] += 1
        }
        
        // const key = counts.join(',');
        
        if (!groups[counts]) {
            groups[counts] = [];
        }
        groups[counts].push(str);
    });
    
    return Object.values(groups);
};